### PR TITLE
Text input verification

### PR DESCRIPTION
**Pull request by Armetiis**
### Key changes
Introduced a new atom component called ConditionCheckingInputText. This component:
- Wraps the existing InputText component
- Validates input's length against given bounds
- Displays an appropriate error message for invalid input

The InputText component has been enhanced with the following features:
- isError argument: Highlights the input field in red if validation fails
- hasClearIcon argument: When true, displays a button to clear the user's input

The new ConditionCheckingInputText has been integrated into relevant screens
"Save" and "Submit" buttons are now disabled when input validation fails
The affected screens are
- AttributePicker
- EditProfile
- AddReview

No error vs with error:

image
image

### Coverage
image

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/cyrcle/ui/addParking/AddScreensNavigationTest.kt [hunk 1]
```diff
@@ -66,7 +67,7 @@ class AddScreensNavigationTest {
         addressViewModel,
         mockedAuthenticator,
         LocalContext.current as android.app.Activity)
-    return listOf<Any>(
+    return listOf(
```
Comment (LouisJamesVasseur) [inline app/src/androidTest/java/com/github/se/cyrcle/ui/addParking/AddScreensNavigationTest.kt:70]: Does this change anything in practice?

File: app/src/androidTest/java/com/github/se/cyrcle/ui/review/ReviewScreenTest.kt [hunk 2]
```diff
@@ -75,7 +75,7 @@ class ReviewScreenTest {
     composeTestRule
         .onNodeWithTag("TopAppBarTitle")
         .assertIsDisplayed()
-        .assertTextContains("Add Your Review")
+        .assertTextContains("Add your review")
```
Comment (LouisJamesVasseur) [inline app/src/androidTest/java/com/github/se/cyrcle/ui/review/ReviewScreenTest.kt:78]: Why remove the captialization? It's a title...

**Review by LouisJamesVasseur — APPROVED** [review]
From both a UI and code perspective, this is a clear and robust implementation. Well done!

File: app/src/androidTest/java/com/github/se/cyrcle/ui/review/ReviewScreenTest.kt (hunk 2 above)
Comment (Armetiis) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/review/ReviewScreenTest.kt:78]: It's true. I just tried removing some inconsistent caps in the strings ressources without really overthinking it. Do you really think it's ok to have "Add Your Review"? To me it looks illegal. Anyways we can come back to these details later, let's keep moving forward witht the important stuff!

Comment (LouisJamesVasseur) [reply app/src/androidTest/java/com/github/se/cyrcle/ui/review/ReviewScreenTest.kt:78]: Completely agree!

**Pull request merged**